### ISSUE TITLE
Should.Throw works incorrectly for TimeoutException

### ISSUE THREAD

**Issue by steck** [shouldly/shouldly#363]
Following test fails

```
[Test]
public void ShouldHandleTimeoutException()
{
    Func<Task> func = async () => { throw new TimeoutException(); };

    Should.Throw<TimeoutException>(func);
}
```

with following message

```
Shouldly.ShouldCompleteInException : 
    Task
        should complete in
    00:00:10
        but did not
  ----> System.TimeoutException : The operation has timed out.
```

**Comment by AdityaTulasi** [comment]
I am getting an exception saying shouldThrow method should throw an exception of type System.Exception but did not throw

**Comment by bmpj13** [comment]
I've been checking out this bug. The problem is that TimeoutException has a special meaning inside Shouldly's.
Should.Throw catches TimeoutException to tell it couldn't run the task in the set time.
An easy fix, I think, would be creating Shouldly's own exception timeouts, so it wouldn't get mixed with the ones available to any user.

Let me know what you think, as I can try to work on it.

**Comment by eliziario** [comment]
looks like this could be closed, as it seems to me that #363 fixes it.

**Comment by JosephWoodward** [comment]
Oh, thanks for point that out @eliziario!
